profile.md

```diff
@@ -660,9 +660,11 @@ If an entity is not entitled to a capability, the scope requested may be ignored
 
 An entity may be entitled to a capability due to membership in a group or entitlement to use a role. The entity may be a member of multiple groups (VOs), with multiple roles, supported by a common implementation (analogous to how VOMS-Admin is operated at CERN). To support this scenario, a `wlcg.capabilityset` value may be included in the scope request to specify the group/role context for the scope request. This can determine the resulting `iss` and `scope` claims in the issued token.
 
+Only one `wlcg.capabilityset` should be in a single request.  If there are additional capability scopes in the request with the same name as those in the capability set, they will be used to limit the capability paths requested.  There is no provision for a requester to completely remove a capability from a capability set, but in that case a different capability set can be defined.
```

Comment:
Could we change "they will be used to limit the capability paths requested" into something like "the effective capabilities for the request will be the intersect of the capability scopes in the capabilityset and the additional capability scopes". I think that would be more deterministic.
Or do you mean that the effective capability scopes are those that are in the set and *not* in the additional scopes? I would be very much against that as it would completely revert the meaning of explicit capability scopes.

Reply:
I was only talking about scopes "with the same name", meaning that they are both in the request and in the capabilityset.   I think it confuses things to talk about additional scopes; those will still behave as normal and request additional scopes.   Does that clarify it?

Reply:
I think I know what you want to say, but it's not really what it says now.
Explicit scopes such as `storage.write:/dune/data` which are *also* in a `wlcg.capabilityset` effectively fully override the latter, right? So I think we mean the following:
> if a `wlcg.capabilityset` scope is requested together with one of its capability scopes, that `wlcg.capabilityset` is ignored.

Or do I still misunderstand?
I do think this will lead to confusion: you ask more things and get less.

Reply:
You're right, I hadn't thought of it that way and it's a simpler way to look at it.  I updated the doc to show that.

Reply:
Two more remarks on the updated text: you say "the scope in the capability set will be ignored", but it's effectively the whole capability set that is ignored, right (apart from the one that is also mentioned explicitly) ? I.e. the result is the same as when the requester would not have requested the `wlcg.capabilityset` at all, but just the capability scopes explicitly?
Perhaps formulate as something like "in other words, when the capabilityset in that request is left out, the result would be the same".
Additionally, it is I think also clearer to rephrase "completely remove a capability from a capability set" with "remove an individual capability from a capability set".

Reply:
> you say "the scope in the capability set will be ignored", but it's effectively the whole capability set that is ignored, right (apart from the one that is also mentioned explicitly) ? I.e. the result is the same as when the requester would not have requested the `wlcg.capabilityset` at all, but just the capability scopes explicitly?

No, that's not what I had in mind.  As is shown in the 4th example, the capabilityset may contain multiple scopes, but only the ones explicitly requested are overridden.

> it is I think also clearer to rephrase "completely remove a capability from a capability set" with "remove an individual capability from a capability set".

Ok, done.  Thanks.

Reply:
Ok, I think I now (finally) understand, so you mean that all capabilities of the same type (`storage.read` in this case) in the `wlcg.capabilityset` are overridden (ignored actually).
And that if I need the capabilityset `wlcg.capabilityset:/dune/dunepro` and also read access for `storage.read:/home/jdoe` I cannot do that without defining a new wlcg.capabilityset (which I cannot do as client) or specifying also all the storage.read-type scopes from the capabilityset explicitly?

Reply:
Yes.

Reply:
Ok, good.
In that case, I suggest using "of the same type" instead of "with the same name". Or more explicitly "of the same type of authorization activity" but that's rather long.
"The same name" to me implies either the whole thing or its path, so something like `storage.read:/dune/data` or just `/dune/data` but not just the type.
For the whole sentence I would change to
"If additionally a scope is requested of the same type as one or more scopes in the capability set, the explicitly requested scope will take precedence; in other words, any scope of that specific type in the capability set will be ignored."

Reply:
Ok I comitted your wording except I said "a scope" instead of "one or more scopes".

Reply:
Thanks!
Sorry to be so picky, and I should have already said this yesterday, but how about also adding
"(such as `storage.read`)"
after "the same type", to leave no room for confusion with the reader.
Other than that, looks fine to me. We should wait for feedback from others though.